content/en/tracing/guide/add_span_md_and_graph_it.md

```diff
@@ -0,0 +1,287 @@
+
+```python
+from ddtrace import tracer
+
+@app.route('/customer/<int:customer_id>')
+@tracer.wrap()
+def handle_customer(customer_id):
+  # get the active span in the context, put there by tracer.wrap()
+  current_span = tracer.current_span()
+  if current_span:
+    current_span.set_tag('customer.id', customer_id)
+```
+
+{{% /tab %}}
+{{% tab "Ruby" %}}
+
+Add metadata directly to `Datadog::Span` objects by calling `#set_tag`:
+
+```ruby
+# An example of a Sinatra endpoint,
+# with Datadog tracing around the request.
+get '/posts' do
+  Datadog.tracer.trace('web.request') do |span|
+    span.set_tag('http.url', request.path)
+  end
+end
+```
+
+Access the current active span from any method within your code. **Note**: If the method is called and there is no span currently active, `active_span` is `nil`.
+
+```ruby
+# e.g. adding tag to active span
+
+current_span = Datadog.tracer.active_span
+current_span.set_tag('<TAG_KEY>', '<TAG_VALUE>') unless current_span.nil?
+```
+
+{{% /tab %}}
+{{% tab "Go" %}}
+
+Add metadata directly to a `Span` interface by calling `SetTag`:
+
+```go
+package main
+
+import (
+    "log"
+    "net/http"
+
+    "gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+func handler(w http.ResponseWriter, r *http.Request) {
+    // Create a span for a web request at the /posts URL.
+    span := tracer.StartSpan("web.request", tracer.ResourceName("/posts"))
+    defer span.Finish()
+
+    // Set tag
+    span.SetTag("http.url", r.URL.Path)
+}
+
+func main() {
+    tracer.Start(tracer.WithServiceName("<SERVICE_NAME>"))
+    defer tracer.Stop()
+    http.HandleFunc("/posts", handler)
+    log.Fatal(http.ListenAndServe(":8080", nil))
+}
+```
+
+Datadog's integrations make use of the `Context` type to propagate the current active span.
+If you want to add span metadata via a tag attached to a `Context`, call the `SpanFromContext` function:
+
+```go
+package main
+
+import (
+    "net/http"
+
+    "gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+func handler(w http.ResponseWriter, r *http.Request) {
+    // Retrieve a span for a web request attached to a Go Context.
+    if span, ok := tracer.SpanFromContext(r.Context()); ok {
+        // Set tag
+        span.SetTag("http.url", r.URL.Path)
+    }
+}
+```
+
+{{% /tab %}}
+{{% tab "Node.js" %}}
+
+
+Add metadata directly to span objects by calling `setTag` or `addTags`:
+
+```javascript
+// An example of an Express endpoint,
+// with Datadog tracing around the request.
+app.get('/posts', (req, res) => {
+  const span = tracer.startSpan('web.request')
+
+  span.setTag('http.url', req.url)
+  span.addTags({
+    'http.method': req.method
+  })
+})
+```
+
+Access the current active span from any method within your code. **Note**: If the method is called and there is no span currently active, `tracer.scope().active()` returns `null`.
+
+```javascript
+// e.g. adding tag to active span
+
+const span = tracer.scope().active()
+
+span.setTag('<TAG_KEY>', '<TAG_VALUE>')
```

Comment:
Let's use a real endpoint with this snippet to add the `customer_id` (same comment as before).